Add NCBI Sequence Read Archive support via fusera

gkelly opened the pull request:
This change set adds support for inputs in the [NCBI SRA](https://www.ncbi.nlm.nih.gov/sra) via [fusera](https://github.com/mitre/fusera).  A user has to provide their NGC authentication file and a fusera Docker image in their configuration file like so:

```
filesystems {
  sra {
    class = "cromwell.filesystems.sra.SraPathBuilderFactory"
    docker-image = "<some hosted docker path>/fusera:latest"
    ngc = "bmNiaV9nYXAfiwgAAAAAAAADBcHBDYQgEADAv1XQAGYXcfErUe5x0diCiFESA0Y8/VD8zTzrlXwMDEsoII9usPT5znZSmTqUohaSg5Gay14TbxsluMGOSBuqDEKefvbwCzv3BAAKoexb5uIbjjg7dq/p9mH7A5VTImxjAAAA"
  }
}
```

The `ngc` parameter must contain the base64-encoded NGC credential file.  The one provided here is an example from the NCBI site.

gkelly commented on filesystems/sra/src/main/scala/cromwell/filesystems/sra/SraPathBuilder.scala:
```diff
@@ -0,0 +1,32 @@
+  override def pathWithoutScheme: String = accession + "/" + path
+
+  protected def newPath(nioPath: NioPath): Path = ???
+  protected def nioPath: NioPath = ???
```
Done.

kshakir replied:
Throwing a `java.lang.Error` means shut-down-the-JVM. In this case I would throw an `UnsupportedOperationException` that can be (possibly) caught and logged.

gkelly commented on filesystems/sra/src/test/scala/cromwell/filesystems/sra/SraPathBuilderSpec.scala:
```diff
@@ -0,0 +1,88 @@
+  private case class Good(description: String,
+                          in: String,
+                          accession: String,
+                          path: String) {}
```
Done.

Horneth commented on filesystems/sra/src/main/scala/cromwell/filesystems/sra/SraPathBuilderFactory.scala:
```diff
@@ -0,0 +1,15 @@
+  extends PathBuilderFactory {
+  def withOptions(options: WorkflowOptions)(implicit as: ActorSystem, ec: ExecutionContext): Future[SraPathBuilder] = Future {
+    new SraPathBuilder()
+  }
```
This could be a `Future.successful()` instead since there's no real reason to do it asynchronously
Also if the `SraPathBuilder` does not depend on workflow option we could even have a single instance somewhere in an `object`

gkelly replied:
Good point.  Done.

gkelly commented:
Added a section to the `Google.md` docs.  Also refactored the config handling a little to allow for a default value for the docker-image, and to allow for running without an NGC.

Horneth commented:
LGTM, merging
